src/base/DaemonCreator.cpp

```diff
@@ -35,6 +35,20 @@ int DaemonCreator::create(bool parentExit) {
   /* Success: Let the parent terminate */
   if (pid > 0) exit(EXIT_SUCCESS);
 
+  /* Child process, write pid file */
+  if (childPidFile != "") {
+    int pidFilehandle = open(childPidFile.c_str(), O_RDWR|O_CREAT, 0600);
+    if (pidFilehandle == -1) {
+      LOG(FATAL) << "Error opening pidfile for writing: "<< childPidFile;
+    }
+
+    // Max pid length for x86_64 is 2^22 ~ 4000000
```

Comment:
Let's make this char array huge (like 4096) or use stringstream so we don't have to worry about overflow.

Reply:
Updated, see new commit.